api/src/main/java/org/motechproject/nms/api/web/UserController.java

```diff
@@ -234,6 +234,29 @@ private UserResponse getKilkariResponseUser(Long callingNumber) {
         return kilkariUserResponse;
     }
 
+    private boolean serviceDeplyedInCircle(Service service, Circle circle) {
+        if (circle == null) {
+            return true;
+        }
+
+        List<State> stateList = circle.getStates();
+        if (stateList == null || stateList.isEmpty()) { // No state available
+            return true;
+        }
+
+        if (stateList.size() == 1) { // only one state available
+            return serviceDeployedInUserState(service, stateList.get(0));
+        }
```

Comment:
Maybe.  If so the same should be done in the kilkari version of the code.
In fact maybe I should refactor them.

On Tue, Sep 22, 2015 at 10:50 AM, kosh <notifications@github.com> wrote:

> In api/src/main/java/org/motechproject/nms/api/web/UserController.java
> <https://github.com/motech-implementations/mim/pull/591#discussion_r40118952>
> :
>
> > @@ -234,6 +234,29 @@ private UserResponse getKilkariResponseUser(Long callingNumber) {
> >          return kilkariUserResponse;
> >      }
> >
> > +    private boolean serviceDeplyedInCircle(Service service, Circle circle) {
> > +        if (circle == null) {
> > +            return true;
> > +        }
> > +
> > +        List<State> stateList = circle.getStates();
> > +        if (stateList == null || stateList.isEmpty()) { // No state available
> > +            return true;
> > +        }
> > +
> > +        if (stateList.size() == 1) { // only one state available
> > +            return serviceDeployedInUserState(service, stateList.get(0));
> > +        }
>
> this if check could be collapsed into the for loop below right?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/motech-implementations/mim/pull/591/files#r40118952>.
>